src/fleet_app/client.cr

```diff
@@ -29,5 +29,13 @@ module FleetApp
       HTTP::Client.new(base_url)
         .post(path, headers: HTTP::Headers{"X-Auth-Token" => ""}, body: body)
     end
+
+    # Performs a POST request on the path with a body and authentication.
+    def post_with_auth(path : String, body : String, basic_auth : String)
+      header = HTTP::Headers.new
+      headers = header.add("Authorization", "Basic #{basic_auth}")
```

Comment:
(3) then here, set `headers` depending on whether basic_auth is empty or not

e.g.
```
if basic_auth
  headers = header.add("Authorization", "Basic #{basic_auth}")
else 
  headers = HTTP::Headers{"X-Auth-Token" => ""}
 end
```
  
Then in VSH or gamessite, you set the `basic_auth` equal to the env variable or equal to an empty string depending on your key values.